src/MedTrackerAPI.Tests/Features/Supplies/AddSupplyTests.cs

```diff
@@ -0,0 +1,61 @@
+using FluentValidation.TestHelper;
+using MedTrackerAPI.Features.Supplies;
+using MedTrackerAPI.Infrastructure;
+using static MedTrackerAPI.Tests.FakeFactory;
+
+namespace MedTrackerAPI.Tests.Features.Supplies;
+
+public class AddSupplyTests
+{
+    private Device _device;
+
+    [SetUp]
+    public void Setup()
+    {
+        _device = CreateFakeDevice();
+    }
+    
+    [TestCase(null)]
+    [TestCase("")]
+    public void GivenNoDescription_ThenIsNotValid(string? value) =>
+        new AddSupply.AddSupplyCommandValidator().TestValidate(new AddSupply.AddSupplyCommand
+        {
+            Description = value
+        }).ShouldHaveValidationErrorFor(x => x.Description);
+
+    [TestCase(null)]
+    [TestCase("")]
+    public void GivenNoPartNumber_ThenIsNotValid(string? value) =>
+        new AddSupply.AddSupplyCommandValidator().TestValidate(new AddSupply.AddSupplyCommand
+        {
+            PartNumber = value
+        }).ShouldHaveValidationErrorFor(x => x.PartNumber);
+
+    [TestCase(null)]
+    [TestCase("")]
+    public void GivenNoLotNumber_ThenIsNotValid(string? value) =>
+        new AddSupply.AddSupplyCommandValidator().TestValidate(new AddSupply.AddSupplyCommand
+        {
+            LotNumber = value
+        }).ShouldHaveValidationErrorFor(x => x.LotNumber);
+    
+    [TestCase(0)]
+    public void GivenNoDeviceId_ThenIsNotValid(int value) =>
+        new AddSupply.AddSupplyCommandValidator().TestValidate(new AddSupply.AddSupplyCommand
+        {
+            DeviceId = value
+        }).ShouldHaveValidationErrorFor(x => x.DeviceId);
+    
+    [Test]
+    public void GivenAllValidFields_ThenIsValid()
+    {
+        new AddSupply.AddSupplyCommandValidator().TestValidate(new AddSupply.AddSupplyCommand
+        {
+            Description = _device.WithSupplies().Description,
```

Comment:
I would just do the setup once (WithSupply) and then map it into the object instead of getting different supplies each time.

Reply:
I'm not exactly sure what you mean here. Are you saying

Setup Method 
```
public void Setup()
    {
        _device = CreateFakeDevice();
        _device.WithSupplies();
    }
```
Then in the Test do something like
 ```
new AddSupply.AddSupplyCommandValidator().TestValidate(new AddSupplyCommand
        {
            Description = _device.Supplies.First().Description,
```

Reply:
Nailed it. Exactly.